shared/chat/inbox/container.js

```diff
@@ -78,36 +80,67 @@ const filteredInbox = createImmutableEqualSelector(
 const getRows = createImmutableEqualSelector([filteredInbox, getPending], (inbox, pending) => {
   const [smallIds, bigTeamToChannels] = inbox
 
-  const pids = I.List(pending.keySeq().map(k => ({conversationIDKey: k, teamname: null})))
-  const sids = I.List(smallIds.map(s => ({conversationIDKey: s, teamname: null})))
+  const pids = I.List(pending.keySeq().map(k => ({conversationIDKey: k, type: 'small'})))
+  const sids = I.List(smallIds.map(s => ({conversationIDKey: s, type: 'small'})))
 
-  const teams = I.List(
+  const bigTeams = I.List(
     flatten(
       Object.keys(bigTeamToChannels).sort().map(team => {
         const channels = bigTeamToChannels[team]
         return [
           {
             teamname: team,
+            type: 'bigHeader',
           },
         ].concat(
-          Object.keys(channels)
-            .sort()
-            .map(channel => ({channelname: channel, conversationIDKey: channels[channel].id, teamname: team}))
+          Object.keys(channels).sort().map(channel => ({
+            channelname: channel,
+            conversationIDKey: channels[channel].id,
+            teamname: team,
+            type: 'big',
+          }))
         )
       })
     )
   )
 
-  return pids.concat(sids).concat(teams)
+  const allSmallTeams = pids.concat(sids)
+  return [allSmallTeams, bigTeams]
 })
 
-const mapStateToProps = (state: TypedState, {isActiveRoute}) => ({
-  filter: getFilter(state),
-  isActiveRoute,
-  isLoading: state.chat.get('inboxUntrustedState') === 'loading',
-  rows: getRows(state),
-  showNewConversation: state.chat.inSearch && state.chat.inboxSearch.isEmpty(),
-})
+const mapStateToProps = (state: TypedState, {isActiveRoute, smallTeamsExpanded}) => {
+  const [allSmallTeams, bigTeams] = getRows(state, smallTeamsExpanded)
+
+  let smallTeams = allSmallTeams
+  let showSmallTeamsExpandDivider = false
+  if (bigTeams.count() && allSmallTeams.count() > smallteamsCollapsedMaxShown) {
```

Comment:
figure out if we should show the divider (don't show if there are no big teams or if your small teams list isn't big enough